Library/Homebrew/bottles.rb

```diff
@@ -39,11 +38,19 @@ def bottle_tag
   end
 end
 
-def bottle_filename_formula_name filename
-  path = Pathname.new filename
-  version = BottleVersion.parse(path).to_s
-  basename = path.basename.to_s
-  basename.rpartition("-#{version}").first
+def bottle_resolve_formula_names bottle_file
+  receipt_file_path = Utils.popen_read("tar", "-tzf", bottle_file, "*/*/INSTALL_RECEIPT.json").chomp
```

Comment:
For gnu-tar (hence linuxbrew), it requires an extract flag (`--wildcards`) to make it work. Should I add os detection here,   or put it only in linuxbrew?